beacon-chain/blockchain/core.go

```diff
@@ -480,7 +492,68 @@ func (b *BeaconChain) initCycle(cState *types.CrystallizedState, aState *types.A
 		RecentBlockHashes:   recentBlockHashes,
 	}, aState.GetBlockVoteCache())
 
-	return newCrystallizedState, newActiveState
+	return newCrystallizedState, newActiveState, nil
+}
+
+// processCrosslinks checks if the proposed shard block has recevied
+// 2/3 of the votes. If yes, we update crosslink record to point to
+// the proposed shard block with latest dynasty and slot numbers.
+func (b *BeaconChain) processCrosslinks(
+	crosslinkRecords []*pb.CrosslinkRecord,
+	validators []*pb.ValidatorRecord,
+	pendingAttestations []*pb.AttestationRecord,
+	dynasty uint64,
+	slot uint64) ([]*pb.CrosslinkRecord, error) {
+
+	type shardBlockHash struct {
+		id        uint64
+		blockHash []byte
+	}
+	shardBlockHashBalance := make(map[*shardBlockHash]uint64)
```

Comment:
Hmm, I don't think this works as expected. This uses the pointer as the key, which means that two different `shardBlockHash` pointers with the same value are considered different keys. You can tell that this can't work because go doesn't support slice comparison, and the golang compiler complains if you change `shardBlockHashBalance` to `map[shardBlockHash]uint64` (no pointer). 

I'll put in a fix as part of #473.

Reply:
Hmm I'm not sure what the intention of this map is. It seems to not be used for anything? Was the original intention to ensure that if there's multiple attestations for the same shard id/hash combo, to calculate the sum?

Reply:
Yup. The intention to keep track of the sum for shard id/hash combo.

Reply:
I just looked at the python implementation, I'm assuming that was the intention. I'll fix this in #473!

Reply:
Thanks!